modules/planning/tasks/dp_poly_path/trajectory_cost.cc

```diff
@@ -137,37 +136,37 @@ ComparableCost TrajectoryCost::CalculatePathCost(
     const double end_s, const uint32_t curr_level, const uint32_t total_level) {
   ComparableCost cost;
   double path_cost = 0.0;
-  for (double path_s = 0.0; path_s < (end_s - start_s);
-       path_s += config_.path_resolution()) {
-    const double l = curve.Evaluate(0, path_s);
+  std::function<double(const double)> quasi_softmax = [this](const double x) {
+    const double l0 = this->config_.path_l_cost_param_l0();
+    const double b = this->config_.path_l_cost_param_b();
+    const double k = this->config_.path_l_cost_param_k();
+    return (b + std::exp(-k * (x - l0))) / (1.0 + std::exp(-k * (x - l0)));
+  };
 
-    std::function<double(const double)> quasi_softmax = [this](const double x) {
-      const double l0 = this->config_.path_l_cost_param_l0();
-      const double b = this->config_.path_l_cost_param_b();
-      const double k = this->config_.path_l_cost_param_k();
-      return (b + std::exp(-k * (x - l0))) / (1.0 + std::exp(-k * (x - l0)));
-    };
+  const auto &vehicle_config =
+      common::VehicleConfigHelper::instance()->GetConfig();
+  const double width = vehicle_config.vehicle_param().width();
+
+  for (double curve_s = 0.0; curve_s < (end_s - start_s);
+       curve_s += config_.path_resolution()) {
+    const double l = curve.Evaluate(0, curve_s);
 
     path_cost += l * l * config_.path_l_cost() * quasi_softmax(std::fabs(l));
 
     double left_width = 0.0;
     double right_width = 0.0;
-    reference_line_->GetLaneWidth(path_s, &left_width, &right_width);
-
-    const auto &vehicle_config =
-        common::VehicleConfigHelper::instance()->GetConfig();
-    const double width = vehicle_config.vehicle_param().width();
+    reference_line_->GetLaneWidth(curve_s, &left_width, &right_width);
```

Comment:
curve_s should be curve_s + start_s ?

Reply:
Maybe it is better to keep curve_s in this change and fix it later, so we know nothing affects the tests.

Reply:
That's the plan